acvm-repo/acvm/tests/solver.rs

```diff
@@ -722,3 +724,531 @@ fn memory_operations() {
+    ys: Vec<(FieldElement, bool)>,
+) -> Vec<FieldElement> {
+    bigint_solve_binary_op_opt(Some(middle_op), modulus, xs, ys)
+}
+
+fn bigint_solve_from_to_le_bytes(
+    modulus: Vec<u8>,
+    input: Vec<(FieldElement, bool)>,
+) -> Vec<FieldElement> {
+    bigint_solve_binary_op_opt(None, modulus, input, vec![]) // TODO: input instead of vec![] ?
+}
+
+// NOTE: an "average" bigint is large, so consider increasing the number of proptest shrinking
+// iterations (from the default 1024) to reach a simplified case, e.g.
+// PROPTEST_MAX_SHRINK_ITERS=1024000
+proptest! {
+    #[test]
+    fn bigint_from_to_le_bytes_zero_one(modulus in select(allowed_bigint_moduli()), zero_or_ones_constant: bool, use_constant: bool) {
+        let zero_function_input = if zero_or_ones_constant {
+            FieldElement::one()
+        } else {
+            FieldElement::zero()
+        };
+        let zero_or_ones: Vec<_> = modulus.iter().map(|_| (zero_function_input, use_constant)).collect();
+        let expected_results = drop_use_constant(&zero_or_ones);
+        let results = bigint_solve_from_to_le_bytes(modulus.clone(), zero_or_ones);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    fn bigint_from_to_le_bytes((input, modulus) in bigint_with_modulus()) {
+        let expected_results: Vec<_> = drop_use_constant(&input);
+        let results = bigint_solve_from_to_le_bytes(modulus.clone(), input);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    // TODO(https://github.com/noir-lang/noir/issues/5580): desired behavior?
+    fn bigint_from_to_le_bytes_extra_input_byte((input, modulus) in bigint_with_modulus(), extra_byte: u8, use_constant: bool) {
+        let mut input = input;
+        input.push((FieldElement::from(extra_byte as u128), use_constant));
+        let expected_results: Vec<_> = drop_use_constant(&input);
+        let results = bigint_solve_from_to_le_bytes(modulus.clone(), input);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    // TODO(https://github.com/noir-lang/noir/issues/5580): desired behavior?
+    fn bigint_from_to_le_bytes_two_extra_input_bytes((input, modulus) in bigint_with_modulus(), extra_byte: u8, extra_byte_2: u8, use_constant: bool, use_constant_2: bool) {
+        let mut input = input;
+        input.push((FieldElement::from(extra_byte as u128), use_constant));
+        input.push((FieldElement::from(extra_byte_2 as u128), use_constant_2));
+        let expected_results: Vec<_> = drop_use_constant(&input);
+        let results = bigint_solve_from_to_le_bytes(modulus.clone(), input);
+        prop_assert_eq!(results, expected_results)
+    }
+
+    #[test]
+    // TODO(https://github.com/noir-lang/noir/issues/5580): desired behavior?
+    fn bigint_from_to_le_bytes_extra_input_bytes((input, modulus) in bigint_with_modulus(), extra_bytes_len: u8, extra_bytes in proptest::collection::vec(any::<(u8, bool)>(), u8::MAX as usize)) {
```

Comment:
why do you have bigint_from_to_le_bytes_extra_input_byte and bigint_from_to_le_bytes_two_extra_input_bytes, instead of using bigint_from_to_le_bytes_extra_input_bytes ?

Reply:
I had written these to ensure the random generator hits the specific `1..2` byte cases, since the default `proptest` strategy [generates `u8`'s within a uniform interval](https://docs.rs/proptest/latest/src/proptest/num.rs.html#47).

But since the general case is passing, I've removed the other two tests.